crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -601,13 +606,29 @@ impl CheckpointBuilder {
 
             let epoch_rolling_gas_cost_summary =
                 self.get_epoch_total_gas_cost(last_checkpoint.as_ref().map(|(_, c)| c), &effects);
+
+            self.accumulator.accumulate_checkpoint(
+                effects.clone(),
+                sequence_number,
+                self.epoch_store.clone(),
+            )?;
+
             if last_checkpoint_of_epoch {
                 self.augment_epoch_last_checkpoint(
                     &epoch_rolling_gas_cost_summary,
                     timestamp_ms,
                     &mut effects,
                 )
                 .await?;
+
+                let root_state_digest = Some(
+                    self.accumulator
```

Comment:
Can you add `monitored_scope` here to measure time `digest_epoch` takes to execute?